lib/gateways/google_drive.rb

```diff
@@ -0,0 +1,32 @@
+require "google/apis/drive_v3"
+
+module Gateways
+  class GoogleDrive
+    SCOPE = "https://www.googleapis.com/auth/drive".freeze
+
+    def initialize(credentials:)
+      authorizer = Google::Auth::ServiceAccountCredentials.make_creds(
+        json_key_io: StringIO.new(credentials),
+        scope: SCOPE,
+      )
+
+      @service = Google::Apis::DriveV3::DriveService.new
+      @service.authorization = authorizer
+    end
+
+    def write(file_name:, folder_name:, upload_source:, mime_type: "text/plain")
+      file = Google::Apis::DriveV3::File.new
+      file.name = file_name
+      file.mime_type = mime_type
+
+      drive_file = @service.create_file(file, upload_source: upload_source)
+
+      if folder_name
```

Comment:
In what circumstances would `folder_name` be not truthy?

Reply:
If it were not provided... the default is to write the file to the root of the account, if you don't then assign in to a folder, it stays there.

In this specific use, a folder will always be provided, but the gateway could feasibly be reused and not.

Reply:
OK, thanks :+1: